controllers/apply-next/joi-extensions/date-range.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+const { fromDateParts } = require('../../../modules/dates');
+
+module.exports = function dateParts(joi) {
+    return {
+        name: 'dateRange',
+        base: joi.object({
+            start: {
+                day: joi
```

Comment:
You could probably make this a variable, e.g.

```js
const dateParts = {
    day: joi
        .number()
        .integer()
        .required(),
    month: joi
        .number()
        .integer()
        .required(),
    year: joi
        .number()
        .integer()
        .required()
};
...
joi.object({  start: dateParts, end: dateParts })
```

Reply:
😊 good call!